### ISSUE TITLE
extreme lag when loading lists

### ISSUE THREAD

**Issue by smenor** [bluesky-social/social-app#6947]
### Steps to Reproduce

1. click on profile
2. click on add to list

Expected result: lists appear quickly

Actual result: lists appear but show loading for at least 15-20 seconds

### Attachments

<img width="551" alt="Screenshot 2024-12-04 at 09 25 32" src="https://github.com/user-attachments/assets/9219b9f6-a4e7-4373-bc50-93a121561195">

### What platform(s) does this occur on?

iOS, Web (Desktop), Web (Mobile)

### Device Info

iOS 15.2; macOS 15.2 Beta (24C5089c)

### What version of the app are you using?

Build version: 1.95.0.634; Bundle info: 2a575ad (prod); Bundle date: 24112703; Platform: ios; Platform version: 18.2

### Additional Information

This has been going on for many months at least and is incredibly annoying

**Comment by Sininini** [comment]
Yeah I can confirm. It had been really annoying me for a long time since I very frequently add accounts to lists.
It happens on Android app and in both Windows Chrome and ios Safari.

**Comment by xarielml** [comment]
This is still an issue. Video shows it takes >12 seconds to load upon first opening Bluesky. Video is on iPhone 14 Pro.

Build version: 1.102.0.1031; Bundle info: 1cdbfc709 (prod); Bundle date: 25051623; Platform: ios; Platform version: 18.4.1; Anonymous ID: c653370f-42c4-48e2-8702-958d53f9adcd

https://github.com/user-attachments/assets/704d5352-d780-4d72-9e9d-24b4f7e6a8ef

**Comment by hangukhiphop** [comment]
I was going to submit a ticket for this, but I'm glad to see someone beat me to the cut. From a dev perspective, this seems like the simple issue of not "lazy loading" the lists. Essentially, that would mean loading only the list titles and then an individual list's contents once you select it.

**Comment by smenor** [comment]
@hangukhiphop that may be a factor too but I think they've also under-resourced the endpoint where lists are getting pulled from and it's having absurdly long response times ( though I haven't tried pulling directly from atproto recently and can't say with confidence )

**Comment by therichferro** [comment]
This issue is still going on and it's taking a really long time for some accounts with 20+ lists, I have an account I use to curate feeds and it's not uncommon to be waiting over 1 minute for the list "add" button to finally load.

**Comment by smenor** [comment]
It's really hard to imagine that this isn't deliberate
